pkg/keys/ed25519.go

```diff
@@ -83,8 +83,20 @@ func GenerateEd25519Key() (*ed25519Signer, error) {
 	}, nil
 }
 
-func (e *ed25519Signer) SignMessage(message []byte) ([]byte, error) {
-	return e.Sign(rand.Reader, message, crypto.Hash(0))
+func (e *ed25519Signer) SignMessage(message []byte) ([]data.Signature, error) {
+	sig, err := e.Sign(rand.Reader, message, crypto.Hash(0))
+	if err != nil {
+		return nil, err
+	}
+	ids := e.PublicData().IDs()
+	sigs := make([]data.Signature, 0, len(ids))
+	for _, id := range ids {
```

Comment:
Do all of these ids have the same signature? Or should `sig` be calculated in this loop?

Reply:
Yep, all the same. These are the IDs of the key, e.g. maybe computed by two hashes. We should get rid of this though. https://github.com/theupdateframework/go-tuf/issues/158